### PR TITLE
Add release version to function app template

### PR DESCRIPTION
**Pull request by pownkel**
#### Description of changes

Add release version environment variable when deploying function app template instead of in a seperate deployment step.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/resource-deployment/scripts/function-app-create.sh [hunk 1]
```diff
@@ -67,6 +68,13 @@ installAzureFunctionsCoreTools() {
     esac
 }
 
+setReleaseVersionInTemplate() {
+    packageName=$1
+    templateFilePath="${0%/*}/../templates/function-$packageName-app-template.json"
+    parsedTemplateFilePath="${0%/*}/../templates/function-$packageName-app-template-parsed.json"
+    sed -e "s@%RELEASE_VERSION_NUMBER%@$releaseVersion@" "$templateFilePath" >"$parsedTemplateFilePath"
```
Comment (lamaks) [inline packages/resource-deployment/scripts/function-app-create.sh]: we just need to pass value as parameter like below
`--parameters clientId="$webApiAdClientId" namePrefix="$functionAppNamePrefix" releaseVersion=... `

File: packages/resource-deployment/templates/function-web-api-app-template.json [hunk 2]
```diff
@@ -113,6 +113,10 @@
                         {
                             "name": "KEY_VAULT_URL",
                             "value": "[concat('https://', parameters('keyVaultName'), '.vault.azure.net/')]"
+                        },
```
Comment (lamaks) [inline packages/resource-deployment/templates/function-web-api-app-template.json]: add required parameter `releaseVersion` to `parameters` section

File: packages/resource-deployment/templates/function-web-workers-app-template.json [hunk 3]
```diff
@@ -110,6 +110,10 @@
                         {
                             "name": "WEB_API_BASE_URL",
                             "value": "[parameters('webApiBaseUrl')]"
+                        },
```
Comment (lamaks) [inline packages/resource-deployment/templates/function-web-workers-app-template.json]: add required parameter `releaseVersion` to `parameters` section

File: packages/resource-deployment/scripts/install.sh [hunk 4]
```diff
@@ -80,11 +80,12 @@ while getopts ":r:s:l:e:o:p:c:t:" option; do
     p) publisherEmail=${OPTARG} ;;
     c) webApiAdClientId=${OPTARG} ;;
     t) webApiAdClientSecret=${OPTARG} ;;
+    v) releaseVersion=${OPTARG} ;;
     *) exitWithUsageInfo ;;
     esac
 done
 
-if [[ -z $resourceGroupName ]] || [[ -z $subscription ]] || [[ -z $environment ]] || [[ -z $orgName ]] || [[ -z $publisherEmail ]] || [[ -z $webApiAdClientId ]] || [[ -z $webApiAdClientSecret ]]; then
+if [[ -z $resourceGroupName ]] || [[ -z $subscription ]] || [[ -z $environment ]] || [[ -z $orgName ]] || [[ -z $publisherEmail ]] || [[ -z $webApiAdClientId ]] || [[ -z $webApiAdClientSecret ]] || [[ -z $releaseVersion ]]; then
```
Comment (haonliu) [inline packages/resource-deployment/scripts/install.sh]: We are going to pass the predefined release id right?

Comment (pownkel) [reply packages/resource-deployment/scripts/install.sh]: yes, the release id from the pipeline

**Pull request merged**